var/spack/repos/builtin/packages/armadillo/package.py

```diff
@@ -46,18 +46,20 @@ class Armadillo(Package):
     depends_on('hdf5', when='+hdf5')
 
     def install(self, spec, prefix):
+        arpack = find_libraries(['libarpack'], root=spec[
+                                'arpack-ng'].prefix.lib, shared=True)
+        superlu = find_libraries(['libsuperlu'], root=spec[
+                                 'superlu'].prefix, shared=False, recurse=True)
```

Comment:
I would call these variables arpack_libs and superlu_libs just to match blas_libs and lapack_libs.